Sources/momento/internal/PubsubClient.swift

```diff
@@ -102,29 +108,20 @@ class PubsubClient: PubsubClientProtocol {
         } catch let err as GRPCStatus {
             return TopicPublishError(error: grpcStatusToSdkError(grpcStatus: err))
         } catch {
-            return TopicPublishError(error: UnknownError(message: "unknown error"))
+            return TopicPublishError(error: UnknownError(message: "unknown publish error \(error)"))
         }
     }
     
-    func subscribe() async throws -> String {
+    func subscribe(cacheName: String, topicName: String) async throws -> TopicSubscribeResponse {
         var request = CacheClient_Pubsub__SubscriptionRequest()
-        request.cacheName = "not-a-cache"
-        request.topic = "bar"
-        let result = self.client.subscribe(request)
-        
-        print("About to print items loop:")
-        do {
-            for try await item in result {
-                print("Item:", item)
-            }
-        } catch let myError as GRPCStatus {
-            throw grpcStatusToSdkError(grpcStatus: myError)
-        } catch {
-            print("Caught error while looping:", error)
-        }
+        request.cacheName = cacheName
+        request.topic = topicName
         
-        return "calling client.subscribe"
+        let result = self.client.subscribe(request)
+        return TopicSubscribeSuccess(subscription: result)
     }
     
-    
+    func close() {
+        let _ = self.client.channel.close()
```

Comment:
This closes the whole GRPC channel, closing the entire TopicClient. Haven't yet found a way to close only specific subscription streams, but I can remove this for now if we don't want to include it